File: backend/QuizHubApplication/DTOs/Responses/UserQuizAttemptDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Responses
{
    public record UserQuizAttemptDto(
        int QuizAttemptId,
        int QuizId,
        string Title,
        int TimeLimit,
        List<UserQuizAttemptQuestionDto> Questions
        )
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/quiz-attempts/model/UserQuizAttemptDto.ts
import { UserQuizAttemptQuestionDto } from './UserQuizAttemptQuestionDto';

export interface UserQuizAttemptDto {
    quizAttemptId: number;
    quizId: number;
    title: string;
    timeLimit: number;
    questions: UserQuizAttemptQuestionDto[];
}
